### PR TITLE
[13.x] Use unique IDs on client model

### LINKED ISSUE #1706: Hope to support ULID
Hope to support ULID

### PR DESCRIPTION
**Pull request by hafezdivandari**
Uses `\Illuminate\Database\Eloquent\Concerns\HasUniqueIds` trait's methods on `Client` model instead of the old way of generating / assigning unique ID to primary key. It's now much easier to override `newUniqueId` and use ULID for example (fixes #1706) also avoid the issues like #1491.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -59,19 +59,16 @@ class Client extends Model
...
-        parent::boot();
+        parent::__construct($attributes);
 
-        static::creating(function ($model) {
-            if (Passport::clientUuids()) {
-                $model->{$model->getKeyName()} = $model->{$model->getKeyName()} ?: (string) Str::orderedUuid();
-            }
-        });
+        $this->usesUniqueIds = Passport::clientUuids();
     }
```
Comment (driesvints) [inline src/Client.php:72]: We don't recommend overwriting a model's constructor. Can we do this somehow else? maybe with a `bootingX` method or something?

Comment (hafezdivandari) [reply src/Client.php:72]: boot methods are static but `$this->usesUniqueIds` is not, we can use `initializeX` method, are you ok with defining a new trait for this? tbh, I personally prefer to get rid of `Passport::clientUuids()` totally.

Comment (driesvints) [reply src/Client.php:72]: We can't really get rid of that method unless we require older apps to fill out that database column which is quite a bit of a hassle for them. I guess we can give this one a shot.

Comment (hafezdivandari) [reply src/Client.php:72]: @driesvints actually there is no need to fill out that DB column, `string(36)` can be used if they were using incremental ids, in this case they can keep old ids, and get uuid for new clients. what you think?

Comment (hafezdivandari) [reply src/Client.php:72]: > unless we require older apps to fill out that database column

Not necessarily, they can keep the old IDs as is, and get uuid for new clients by using `string(36)` column type.

**Pull request merged**